crates/polars-sql/src/functions.rs

```diff
@@ -860,13 +860,21 @@ impl SQLFunctionVisitor<'_> {
             #[cfg(feature = "nightly")]
             InitCap => self.visit_unary(|e| e.str().to_titlecase()),
             Left => self.try_visit_binary(|e, length| {
-                Ok(e.str().slice(lit(0), match length {
-                    Expr::Literal(LiteralValue::Int64(n)) => lit(n as u64),
+                Ok(match length {
+                    Expr::Literal(Null) => lit(Null),
+                    Expr::Literal(LiteralValue::Int64(0)) => lit(""),
+                    Expr::Literal(LiteralValue::Int64(n)) => {
+                        let len = if n > 0 { lit(n) } else { (e.clone().str().len_chars() + lit(n)).clip_min(lit(0)) };
+                        e.str().slice(lit(0), len)
+                    },
+                    Expr::Literal(_) => polars_bail!(InvalidOperation: "Invalid 'n_chars' for Left: {}", function.args[1]),
                     _ => {
-                        polars_bail!(InvalidOperation: "Invalid 'length' for Left: {}", function.args[1]);
+                            when(length.clone().gt_eq(lit(0)))
```

Comment:
Good thing we have CSE!

Reply:
Definitely! Mapping back to SQL behaviour was awkward enough without having to worrying about eating a large performance hit too 😅